inc/azure_c_util/thandle.h

```diff
@@ -0,0 +1,189 @@
+                    /*Codes_SRS_THANDLE_02_007: [ If *t1 is NULL and t2 is NULL then THANDLE_ASSIGN shall return. ]*/                                                   \
+                    /*so nothing to do, leave them as they are*/                                                                                                        \
+                }                                                                                                                                                       \
+                else                                                                                                                                                    \
+                {                                                                                                                                                       \
+                    /*Codes_SRS_THANDLE_02_008: [ If *t1 is NULL and t2 is not NULL then THANDLE_ASSIGN shall increment the reference count of t2 and store t2 in *t1. ]*/ \
+                    THANDLE_INC_REF(T)(t2);                                                                                                                             \
+                    * (T const**)t1 = t2;                                                                                                                               \
+                }                                                                                                                                                       \
+            }                                                                                                                                                           \
+            else                                                                                                                                                        \
+            {                                                                                                                                                           \
+                if (t2 == NULL)                                                                                                                                         \
+                {                                                                                                                                                       \
+                    /*Codes_SRS_THANDLE_02_009: [ If *t1 is not NULL and t2 is NULL then THANDLE_ASSIGN shall decrement the reference count of *t1 and store NULL in *t1. ]*/ \
+                    THANDLE_DEC_REF(T)(*t1);                                                                                                                            \
+                    * (T const**)t1 = t2;                                                                                                                               \
```

Comment:
>(T const**) [](start = 22, length = 11)

I was looking for this cast :-)

Reply:
Well - we HAVE to cheat a bit. But "keep it low". Cannot be done other way - or let the handles be assignable - which is somewhat ofd a corner stone of the PR.

---
In reply to: [385373842](https://github.com/Azure/azure-c-util/pull/16#discussion_r385373842) [](ancestors = 385373842)